### ISSUE TITLE
heapster in crashloop because /var/run/secrets/kubernetes.io/serviceaccount/token

### ISSUE THREAD

**Issue by dchen1107** [GoogleCloudPlatform/kubernetes#10511]
I had cluster/kube-down.sh and hack/dev-build-and-up.sh for several times (roughtly 5 times) this afternoon, and ran into this twice:

I noticed that heapster has very high restart count here: 
```
$ cluster/kubectl.sh get pod
NAME                                           READY     REASON    RESTARTS   AGE
fluentd-cloud-logging-kubernetes-minion-d615   1/1       Running   0          59s
fluentd-cloud-logging-kubernetes-minion-ds5p   1/1       Running   0          1m
fluentd-cloud-logging-kubernetes-minion-ed39   1/1       Running   0          1m
fluentd-cloud-logging-kubernetes-minion-srpn   1/1       Running   0          1m
kube-dns-v4-af0jm                              3/3       Running   1          2m
monitoring-heapster-v5-pc3el                   0/1       Running   7          2m
monitoring-influx-grafana-v1-ejij9             2/2       Running   0          2m
```

I checked heapster container's log:

```
I0629 23:39:15.986783       1 heapster.go:52] /heapster --source=kubernetes:'' --sink=gcl --sink=influxdb:http://monitoring-influxdb:8086 --poll_duration=2m --stats_resolution=1m
I0629 23:39:15.986897       1 heapster.go:53] Heapster version 0.15.0
E0629 23:39:15.987040       1 heapster.go:59] open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory
```

cc/ @vmarmol since @vishh is out.

**Comment by dchen1107** [comment]
My cluster is synced to commit: fb9c126fd4a2d4130d7ecc405dd59468c92fe0e0

Merge: 450d36f 8c70fa4
Author: Alex Robinson <alexdwanerobinson@gmail.com>
Date:   Mon Jun 29 14:34:05 2015 -0700

    Merge pull request #10078 from ixdy/jenkins-e2e-gce-configs
    
    Determine e2e test configs for GCE using Jenkins project name

**Comment by quinton-hoole** [comment]
The merge of #10369 at 10:56 AM today updates heapster and correlates precisely with the beginning of the heapster problem seen in e2e, which is the cause of the failed cluster builds today.

**Comment by a-robinson** [comment]
I think that calling it the "cause" is a bit of a stretch since the underlying issue seems to be with the serviceaccount, but reverting it should indeed stop the bleeding.

**Comment by mikedanese** [comment]
I saw also see a restart by heapster with this err then after heapster comes back up it functions normally.
```
I0630 00:19:48.284798       1 heapster.go:52] /heapster --source=kubernetes:'' --sink=gcl --sink=influxdb:http://monitoring-influxdb:8086 --poll_duration=2m --stats_resolution=1m
I0630 00:19:48.284895       1 heapster.go:53] Heapster version 0.15.0
I0630 00:19:48.285424       1 kube_factory.go:166] Using Kubernetes client with master "https://10.0.0.1:443" and version "v1beta3"
I0630 00:19:48.285453       1 kube_factory.go:167] Using kubelet port 10255
W0630 00:19:48.286361       1 request.go:308] field selector: v1beta3 - pods - spec.host - : need to check if this is versioned correctly.
W0630 00:19:48.366361       1 request.go:308] field selector: v1beta3 - pods - spec.host - : need to check if this is versioned correctly.
I0630 00:19:52.036607       1 driver.go:284] creating GCL sink
E0630 00:19:52.036698       1 heapster.go:59] The Google Cloud Logging (GCL) sink failed to start: this process must be running on Google Compute Engine (GCE)  
```

**Comment by liggitt** [comment]
Likely the same issue (and resolution) as https://github.com/GoogleCloudPlatform/kubernetes/issues/10507

**Comment by dchen1107** [comment]
@ liggitt I believe this one and another I filed today: 10513 have the same root cause as #10507

Closed it as dup of #10507

**Comment by bh016088** [comment]
I just downloaded and installed heapster and this issue is still occurring.  I am running my own on-premise kube cluster on bare metal Centos.  The only changes I've made to heapster are to change the "type" from LoadBalancer to NodePort.  I've tried the master and 0.18 branch.

suggestions?

[root@node:~/heapster/deploy/kube-config/influxdb]$ kubectl logs heapster-yw9rw --namespace=kube-system
I1104 17:19:21.053644       1 heapster.go:60] /heapster --source=kubernetes:https://kubernetes --sink=influxdb:http://monitoring-influxdb:8086
I1104 17:19:21.053800       1 heapster.go:61] Heapster version 0.18.3
F1104 17:19:21.053848       1 heapster.go:67] open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory
